server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -456,9 +459,16 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
         indicesAliasesRequestRequestValidators = new RequestValidators<>(
                 actionPlugins.stream().flatMap(p -> p.indicesAliasesRequestValidators().stream()).collect(Collectors.toList()));
 
-        restController = new RestController(headers, restWrapper, nodeClient, circuitBreakerService, usageService);
+        Version minimumRestCompatibilityVersion = getMinimumRestCompatibilityVersion(restCompatPlugins);
+        restController = new RestController(headers, restWrapper, nodeClient, circuitBreakerService, usageService,minimumRestCompatibilityVersion);
     }
 
+    private Version getMinimumRestCompatibilityVersion(List<RestCompatibilityPlugin> restCompatPlugins) {
+        return restCompatPlugins.stream()
+            .map(RestCompatibilityPlugin::minimumRestCompatibilityVersion)
+            .max(Version::compareTo)
```

Comment:
Can we through an exception if there are multiple plugins that implement the interface ?

Reply:
right, that method always return current -1 so older APIs will not be accessible (verison -2 - compat apis present still in  for a short period of time after version bump)